sashimi/utilities.py

```diff
@@ -33,3 +35,36 @@ def get_last_parameters(parameter_queue, timeout=0.0001):
         except Empty:
             break
     return params
+
+
+# TODO: Consider this for ScopeCuisine
+class SpeedyArrayBuffer:
+    """
+    Buffer for large data arrays based on numpy and using ctypes for speedy copy of data
+    """
+
+    def __init__(self, size=None, *args, **kwargs):
+        """
+        Create a data object of the appropriate size.
+        """
+        super().__init__(**kwargs)
+        self.np_array = np.ascontiguousarray(
+            np.empty(size // 2)
+        )
+        self.size = size
+
+    def __getitem__(self, slice):
+        return self.np_array[slice]
+
+    def copy_data(self, address):
```

Comment:
I would rename it into `copy_into` (you are copying the data from the address to the array, not vice-versa, right?)